qlib/Util.qm

```diff
@@ -822,4 +826,71 @@ printf (plural(32, "", "tooth", "teeth"));
     public string sub plural (int count, string base, string singular = "", string plural = "s") {
         return sprintf("%d %s", count, count == 1 ? base + singular : base + plural);
     }
+
+    #! Escapes (backslashes) all non-alphanumeric characters in a string.
+    /** Returns string with all non-alphanumerics backslashed; this is useful if you want to match an arbitrary literal string that may have regular expression metacharacters in it.
+
+        @param  text string to escape / backslash
+        @return a string escaped / backslashed
+
+        @par Example:
+        @code{.py}
+string s = regex_escape ("Gain: $100"); # returns "Gain\\:\\ \\$100"
+        @endcode
+
+        @since Util 1.2
+    */
+    public string sub regex_escape (string text) {
+        return regex_subst (text, '([^[:alnum:]])', '\\$1', Qore::RE_Global);
```

Comment:
please change this to
```
test =~ s/([^[:alnum:]])/\\$1/g;
return test;
```

when you use regex_subst(), the regular expression is parsed & compiled every time the function is called.  With the regex subst operator, it's parsed & compiled once at parse time.  they are functionally equivalent, but the function version should only be used when either the pattern or the target strings are variables.